components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/common/APIMgtLatencySynapseHandler.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.common;
+
+import org.apache.synapse.AbstractSynapseHandler;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
+import org.wso2.carbon.apimgt.gateway.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.tracing.TracingSpan;
+import org.wso2.carbon.apimgt.tracing.TracingTracer;
+import org.wso2.carbon.apimgt.tracing.Util;
+
+import java.util.Map;
+import java.util.UUID;
+
+public class APIMgtLatencySynapseHandler extends AbstractSynapseHandler {
+
+    private TracingTracer tracer;
+
+    public APIMgtLatencySynapseHandler() {
+        tracer = ServiceReferenceHolder.getInstance().getTracingService()
+                .buildTracer(APIMgtGatewayConstants.SERVICE_NAME);
+    }
+
+
+    @Override
+    public boolean handleRequestInFlow(MessageContext messageContext) {
+        if (Util.tracingEnabled()) {
+            org.apache.axis2.context.MessageContext axis2MessageContext =
+                    ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+            Map headersMap =
+                    (Map) axis2MessageContext.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+            TracingSpan spanContext = Util.extract(tracer, headersMap);
+
+            String requestId = UUID.randomUUID().toString();
```

Comment:
Why are generating the requestID here? What is the usage of the request ID?

Reply:
Fixed. Remove Request ID